### ISSUE TITLE
Exploring different semantic segmentation models

### ISSUE THREAD

**Issue by rohanakut** [Shared-Reality-Lab/IMAGE-server#404]
During the CSUN demo, one of the participants had mentioned the capability of [MSeg model](https://paperswithcode.com/paper/mseg-a-composite-dataset-for-multi-domain-1). The model seems to give good results. Hence it is important to test this and do a comparative analysis with the existing model

**Comment by rohanakut** [comment]
@jeffbl  after discussing this with @gp1702, we both think that it might be beneficial to create a preprocessor for mseg. The mseg definitely detects more labels in an image and gives richer localisation information. However, in case where the audio team needs just the largest 5 segments(as is the case with current semantic segment), we would need to compare the accuracy of the current model with mseg.
According to me, just on the basis of the amount of information detected I believe Mseg would give richer information of an image. However, for comparing the accuracy of the detected segments we would need to test on a large number of images for which we would need testing scripts. I believe Riana might be able to help here by collecting a few images and labelling them.

**Comment by jeffbl** [comment]
Unless it is trivial to create the preprocessor, I'd suggest that we meet with Riana when she is available, and come up with a broader plan, since we may want to try other things as well, before going to the preprocessor step. We need a plan to be more rigorous about these decisions.

**Comment by jeffbl** [comment]
Moving to next sprint, since @rianadutta is in the planning stage for more formal evaluations that would help us decide.
@rohanakut can you work with @rianadutta to decide if there is a better way to test some of these than having to necessarily build a full preprocessor? Like some other simple API accessible on the command line that we could use to evaluate different models even before we integrate them?

**Comment by jeffbl** [comment]
@rohanakut I'm moving this to Sep12, but I'm unclear on the status on this. Can you summarize where you're at on this?

**Comment by rohanakut** [comment]
@jeffbl I am assigning this to Riana because we need to create a pipeline to compare docker containers with non-dockerised containers.

**Comment by rohanakut** [comment]
@rianadutta post our conversation it was decided that I need to supply you with a Flask version of the mseg model. Post that you can work on creating a script for comparing the models. I am temporarily reassigning this to myself.

**Comment by jeffbl** [comment]
Moved two sprints ahead as per @rianadutta request

**Comment by rbazin** [comment]
This work item fits into story #572
